src/ol/format/kml.js

```diff
@@ -488,7 +488,7 @@ ol.format.KML.readFlatCoordinates_ = function(node) {
  */
 ol.format.KML.readURI_ = function(node) {
   var s = ol.xml.getAllTextContent(node, false).trim();
-  if (node.baseURI) {
+  if (node.baseURI && node.baseURI !== 'about:blank') {
```

Comment:
Strange that Safari now returns `'about:blank'` as baseURI, but I agree that this is the most pragmatic way to work around the issue.